email-ballerina/Module.md

```diff
@@ -0,0 +1,181 @@
+
+```ballerina
+email:Message|email:Error? emailResponse = popClient->receiveMessage();
+```
+
+#### IMAP4 Client
+
+To receive an email using the IMAP4 protocol, you must first create an `email:ImapClient` object. The code for creating an
+ `email:ImapClient` can be found below.
+
+##### Creating a client
+
+The following code creates an IMAP4 client, which connects to the default port(993) and enables SSL.
+```ballerina
+email:ImapClient|email:Error imapClient = new ("imap.email.com",
+                                               "reader@email.com",
+                                               "pass456");
+```
+
+The port number of the server and/or the SSL support can also be configured by passing the following configurations.
+```ballerina
+email:ImapConfiguration imapConfig = {
+    port: 993,
+    enableSsl: true
+};
+
+email:ImapClient|email:Error imapClient = new ("imap.email.com",
+                                               "reader@email.com",
+                                               "pass456",
+                                                imapConfig);
+```
+
+##### Receiving an email
+Once the `email:ImapClient` is created, emails can be received using the IMAP4 protocol through that client.
+Samples for this operation can be found below.
+
+```ballerina
+email:Message|email:Error emailResponse = imapClient->receiveMessage();
+```
+
+#### POP3 and IMAP Listeners
+
+As POP3 and IMAP4 protocols are similar in the listener use cases, lets consider POP3.
+In order to receive emails one-by-one from a POP3 server, you must first create an `email:PopListener` object.
+The code for creating an `email:PopListener` can be found below.
+
+```ballerina
+listener email:PopListener emailListener = check new ({
+    host: "pop.email.com",
+    username: "reader@email.com",
+    password: "pass456",
+    pollingInterval: 2,
+    port: 995
+});
+```
+
+Once initialized a `service` can listen to the new emails as follows. New emails get received at the `onMessage`
+method and when errors happen `onError` method get called.
+
+```ballerina
+service "emailObserver" on emailListener {
+
+    remote function onMessage(email:Message emailMessage) {
+        io:println("Email Subject: ", emailMessage.subject);
+        io:println("Email Body: ", emailMessage.body);
+    }
+
+    remote function onError(email:Error emailError) {
+        io:println("Error while polling for the emails: " + emailError.message());
+    }
+
+}
+```
+
+For information on the operations, which you can perform with this module, see the **Functions**  below. For examples of the usage of the operation, see the following.
+  * [Send Emails Example](https://ballerina.io/learn/by-example/send-email.html)
+  * [Receive Emails using a client Example](https://ballerina.io/learn/by-example/receive-email-using-client.html)
+  * [Receive Emails using a listener Example](https://ballerina.io/learn/by-example/receive-email-using-listener.html)
```

Comment:
We decided to remove BBEs from the API documentation.

Reply:
Removed.